Pull request for branch

saifrais opened the pull request:
2 minor bug fixes & 2 enhancements:
1) Default header: For Top searches & Rising Searches, the first searched keyword was being considered as a header by default
2) Geo searches were limiting the use of EITHER country codes OR country sub-codes. Enabled choosing country codes along with sub-codes
2) Category Trend download: Earlier, you could not download category trends (without putting a query). Incorporated the ability to leave query blank
3) Choose Google Properties: Have appended a query parameter (gprop) which helps choosing between -1)web, 2) news, 3) youtube and 4) froogle

PMassicotte commented:
Hi there and thank you for this PR.

- Can you add some examples on how to use this feature.

eddelbuettel commented:
Also:  generally preferable to have one PR per distinct topic or fix, and often preferred to discuss first (ie file an issue, get consensus, then submit code).

saifrais commented:
Hi @PMassicotte and @eddelbuettel,
Firstly, thanks for a great repository. I am porting a project from Matlab to R...and this has saved me a lot of time!!
@eddelbuettel: Thanks. That makes sense..will follow the protocol going forward.

@PMassicotte : Here is an example which incorporates all the changes proposed

Example:
x <- gtrendsR(query = **""**, cat = "0-47", geo = **c("IN", "IN-MH")**, ch, **gprop** = "news")

Arguments:
**query**: This was required as a compulsory input. GT (Google Trends) gives category information (in percent terms) in its absence. The patch allows a blank input and returns an indexed version of the category trend (indexed to 100 and converted from percentage to index)

**geo**: Master was unable to take an input of a country and country sub-code. In some cases, we would want to index all the country sub-codes (35 Indian states) to the country code (IN).

**gprop**: The patch appends a new argument gprop, which can help choose a particular trend from a specific Google Property (i.e 'news', 'youtube', 'froogle', 'images', 'web -default')

Bug fix: Top.searches and Rising.searches were taking the first entry as the header by default. Fixed the unexpected behavior.

eddelbuettel commented:
@saifrais You need to clean this up and 

- not have it fail regression tests
- not have it show merge conflicts

for us to look at it.

PMassicotte commented:
@saifrais Also please write some roxygen examples (1-2 is enough to show how to use this feature).

PMassicotte commented on R/gtrends.R:
```diff
@@ -571,7 +586,7 @@ as.zoo.gtrends <- function(x, ...) {
       read.csv(
         textConnection(strsplit(vec[i], "\\\n")[[1]]),
-        skip = 1,
+        skip = 1, header = FALSE,
```
If you do that, you will end up with:

```r
> gtrends(c("NBA","NFL"), geo = "CA")$Top.cities.for.nba
              V1  V2
1           City nba
2       Brampton 100
3        Markham  94
4           Ajax  83
5        Toronto  78
6        Vaughan  76
7  Richmond Hill  76
8    Mississauga  76
9       Richmond  62
10      Waterloo  58
11      Oakville  55
12      Winnipeg  51
13        Surrey  51
14      Hamilton  50
15       Windsor  49
16        Oshawa  44
```
The first row should be treated as columns.

saifrais replied:
You are right about Top.cities...But if we keep the header=TRUE (default), the first rows of the next 2 sections (top.searches and rising.searches) seem to take up the 1st query as the header as well. We will need to differentiate between top searches, rising searches and geographic information...probably using regex as proposed.

Considering the merge conflict as well, let me close this pull request and follow the methodical approach as proposed by @eddelbuettel in which we can propose/discuss one fix/enhancement per PR.
